### PR TITLE
ContainerService now attempts to download requested image if it's was…

### PR DESCRIPTION
**Pull request by agolovnev-godeltech**
… not downloaded yet.

#### CHANGE TYPE

 - [ ] Major
 - [x] Minor
 - [ ] Patch

### CODE REVIEW

File: src/CodeReview.Orchestrator/Services/ContainerService.cs [hunk 1]
```diff
@@ -23,6 +29,21 @@ public class ContainerService : IContainerService
             _nameFactory = nameFactory ?? throw new ArgumentNullException(nameof(nameFactory));
         }
 
+        private async Task ValidateImage(IDockerClient client, string imageName) {
```
Comment (DiamondDragon) [inline src/CodeReview.Orchestrator/Services/ContainerService.cs]: Could we renamed this method to `ValidateImageAsync`?

File: src/CodeReview.Orchestrator/Services/ContainerService.cs [hunk 2]
```diff
@@ -23,6 +29,21 @@ public class ContainerService : IContainerService
...
+            var imageListParams = new ImagesListParameters {
+                MatchName = imageName
+            };
+            // Due to some reasons this method ignores match by image name and returns list of all registered images. Thus it's necessary to perform manual lookup.
+            var matches = await client.Images.ListImagesAsync(imageListParams);
+            var found = false;
+            if (matches != null && matches.Count > 0) {
```
Comment (DiamondDragon) [inline src/CodeReview.Orchestrator/Services/ContainerService.cs]: I would follow formatting approach used for rest of code. E.g. put `{` in separate line`

**Review by DiamondDragon — CHANGES_REQUESTED** [review]

File: src/CodeReview.Orchestrator/Services/ContainerService.cs (hunk 1 above)
Comment (agolovnev-godeltech) [reply src/CodeReview.Orchestrator/Services/ContainerService.cs]: done

File: src/CodeReview.Orchestrator/Services/ContainerService.cs (hunk 2 above)
Comment (agolovnev-godeltech) [reply src/CodeReview.Orchestrator/Services/ContainerService.cs]: done

**Review by DiamondDragon — APPROVED** [review]

**Pull request merged**